bluesky_widgets/models/run_engine_client.py

```diff
@@ -0,0 +1,372 @@
+import time
+
+from bluesky_live.list import ListModel
+from bluesky_live.event import EmitterGroup, Event
+from bluesky_queueserver.manager.comms import ZMQCommSendThreads, CommTimeoutError
+
+
+class PlanItem:
+    def __init__(self, name, args):
+        self._name = name
+        self._args = args
+        self.events = EmitterGroup(
+            source=self,
+            name=Event,
+        )
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if value == self._name:
+            return
+        self._name = value
+        self.events.name(name=value)
+
+    @property
+    def args(self):
+        return self._args
+
+    @args.setter
+    def args(self, value):
+        # TODO Deal with *mutation* (editing) of the args the same way we deal
+        # with mutation of plot styles.
+        if value == self._args:
+            return
+        self._args = value
+        self.events.args(args=value)
+
+
+class PlanQueue(ListModel):
+    pass
+
+
+class PlanHistory(ListModel):
+    pass
+
+
+class BigModel:
```

Comment:
Is this needed? It looks like it isn't used it the demo. Maybe we shouldn't add this until/unless need it (especially since it was evidently hard to think of a good name).

Reply:
I didn't delete any of the preexisting code yet. But `BigModel` could probably be deleted. Regarding the new code, I think there should be some renaming done once the first working prototype is ready.